packages/design-system/src/components/tooltip/index.js

```diff
@@ -121,18 +129,46 @@ function Tooltip({
     [placement]
   );
 
-  const positionArrow = useCallback(() => {
-    const anchorElBoundingBox = anchorRef.current?.getBoundingClientRect();
-    const tooltipElBoundingBox = tooltipRef.current?.getBoundingClientRect();
-    if (!tooltipElBoundingBox || !anchorElBoundingBox) {
-      return;
-    }
-    const delta =
-      getBoundingBoxCenter(anchorElBoundingBox) -
-      getBoundingBoxCenter(tooltipElBoundingBox);
+  // When near the bottom of the viewport and the tooltip is placed on the bottom we want to force the tooltip to the top as to not
+  // cutoff the contents of the tooltip.
+  const positionPlacement = useCallback(
+    ({ offset }) => {
+      if (offset) {
+        const neededVerticalSpace = offset.y + offset.height;
+        // if the tooltip was assigned as bottom we want to always check it
+        if (realPlacement.startsWith('bottom')) {
+          // check to see if there's an overlap with the window edge
+          if (neededVerticalSpace >= window.innerHeight) {
```

Comment:
This is my b, i didn't clean up my wip :) these 3 checks could be combined. 

I'd been looking to see if a reset needed to occur elsewhere in this step process, so i'd separated them out. Now that it's all solved it's unnecessary.

Reply:
Totally.